src/background/requestHandler.js

```diff
@@ -175,11 +178,14 @@ export class RequestHandler extends Component {
       if (extState?.useExitRelays) {
         return relays;
       }
+      if (ProxyUtils.browserProxySettingIsValid(browserProxySettings)) {
+        return relays;
+      }
       return ProxyUtils.getDirectProxyInfoObject();
```

Comment:
If you have a proxy, and we are in {type:direct} land, force use the relays if a proxy is there.